addons/binding/org.openhab.binding.tadoac/src/main/java/org/openhab/binding/tadoac/handler/TadoACHandler.java

```diff
@@ -0,0 +1,216 @@
+            updateStatus(ThingStatus.ONLINE);
+        } catch (Exception e) {
+            e.printStackTrace();
+            logger.error("could not send state", e);
+        } finally {
+            sendFuture = null;
+        }
+
+    }
+
+    private synchronized void refreshSetting() {
+        try {
+            String username = (String) getConfig().get("username");
+            String password = (String) getConfig().get("password");
+            int homeid = ((BigDecimal) getConfig().get("homeid")).intValue();
+            int zoneid = ((BigDecimal) getConfig().get("zoneid")).intValue();
+            if (username == null || password == null) {
+                logger.error("login credentials not configured");
+                updateStatus(ThingStatus.OFFLINE);
+                return;
+            }
+            TadoACConnector connector = new TadoACConnector(username, password);
+            TadoACSetting newSetting = connector.getSetting(homeid, zoneid);
+
+            // Check each part and copy them to the previous state (No loss of data)
+            if (newSetting.getPower() != null) {
+                setting.setPower(newSetting.getPower());
+                updateState(CHANNEL_POWER, newSetting.getPower() == TadoACPower.ON ? OnOffType.ON : OnOffType.OFF);
+            }
+            if (newSetting.getFanSpeed() != null) {
+                setting.setFanSpeed(newSetting.getFanSpeed());
+                int fanSpeed = 1;
+                switch (newSetting.getFanSpeed()) {
+                    case LOW:
+                        fanSpeed = 1;
+                        break;
+                    case MIDDLE:
+                        fanSpeed = 2;
+                        break;
+                    case HIGH:
+                        fanSpeed = 3;
+                        break;
+                    default:
+                        break;
+                }
+                updateState(CHANNEL_FANSPEED, new DecimalType(fanSpeed));
+            }
+            if (newSetting.getMode() != null) {
+                setting.setMode(newSetting.getMode());
+                int mode = 1;
+                switch (newSetting.getMode()) {
+                    case COOL:
+                        mode = 1;
+                        break;
+                    case DRY:
+                        mode = 2;
+                        break;
+                    case FAN:
+                        mode = 3;
+                        break;
+                    default:
+                        break;
+                }
+                updateState(CHANNEL_MODE, new DecimalType(mode));
+            }
+            if (newSetting.getTemperature() != null) {
+                setting.setTemperature(newSetting.getTemperature());
+                updateState(CHANNEL_TEMP, new DecimalType(newSetting.getTemperature().getCelsius()));
+            }
+        } catch (Exception e) {
+            logger.error("error while polling for new state", e);
```

Comment:
This also means that you cannot connect to the thing, right? So you should set it to OFFLINE here.